feat: avm module upgrade for macae

AjitPadhi-Microsoft opened the pull request:
## Description
This pull request introduces three new Bicep modules to support AI project, AI services deployment, and AI Foundry connections, and removes the custom implementation for Container App Environment in favor of a standardized AVM module. The changes enhance modularity, reusability, and support for advanced configuration of Azure AI and container resources.

**New modules for AI resource management:**

* Added `ai-project.bicep` module to create and manage AI Foundry projects within an existing Cognitive Services account, with support for custom descriptions, tags, and outputs for key identifiers and endpoints.
* Added `ai-services-deployments.bicep` module to manage deployments and role assignments for Cognitive Services accounts, supporting a wide range of built-in roles, deployment SKUs, and flexible configuration through parameterized inputs.
* Added `aifp-connections.bicep` module to create AI Foundry project connections to Azure Cognitive Search, including secure API key handling and metadata configuration.

**Container App Environment module simplification:**

* Removed the custom implementation of `container-app-environment.bicep`, which handled environment creation, monitoring, redundancy, and networking logic, in favor of using the AVM public module for managed environments. This reduces code duplication and leverages standardized best practices.
<!--
>Thank you for your contribution !
> Please include a summary of the change and which issue is fixed.
> Please also include the context.
> List any dependencies that are required for this change.

Fixes #123
Closes #123
-->

## Pipeline Reference

<!-- Insert your Pipeline Status Badge below -->

| Pipeline |
| -------- |
|[![avm.ptn.sa.multi-agent-custom-automation-engine](https://github.com/Azure/bicep-registry-modules/actions/workflows/avm.ptn.sa.multi-agent-custom-automation-engine.yml/badge.svg)](https://github.com/Azure/bicep-registry-modules/actions/workflows/avm.ptn.sa.multi-agent-custom-automation-engine.yml)|


## Type of Change

<!-- Use the checkboxes [x] on the options that are relevant. -->

- Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation
- [ ] Update to CI Environment or utilities (Non-module affecting changes)

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] I have updated the module's CHANGELOG.md file with an entry for the next version

<!--  Please keep up to date with the contribution guide at https://aka.ms/avm/contribute/bicep -->

AjitPadhi-Microsoft committed:
Updated avm changes for macae

AjitPadhi-Microsoft committed:
Merge branch 'main' into macae-avm-update

AjitPadhi-Microsoft committed:
updated image tags

AjitPadhi-Microsoft committed:
Merge branch 'macae-avm-update' of https://github.com/Ravikirana-Microsoft/bicep-registry-modules into macae-avm-update

AjitPadhi-Microsoft committed:
validation fix

Commit:
Merge branch 'main' into macae-avm-update

AjitPadhi-Microsoft committed:
Merge branch 'main' into macae-avm-update

Commit:
Merge branch 'main' into macae-avm-update

Commit:
fix: Removed region pair for Cosmos DB and added a replica region param

Commit:
Merge branch 'main' into macae-avm-update

Commit:
Merge branch 'main' into macae-avm-update

Commit:
Merge branch 'main' into macae-avm-update

AjitPadhi-Microsoft commented on utilities/pipelines/staticValidation/psrule/ps-rule.yaml:
```diff
@@ -83,3 +83,4 @@ rule:
     - Azure.AppConfig.PurgeProtect
     - Azure.MySQL.MaintenanceWindow # Must be excluded until https://msdata.visualstudio.com/Database%20Systems/_workitems/edit/2788114 is fixed
     - Azure.AppService.AvailabilityZone # Must disable as the serverfarm premium skus are not available in the AVM subscription. The module is WAF-aligned, just the tests don't validate it
+    - Azure.ContainerApp.MinReplicas # Must disable for Multi-Agent Custom Automation Engine module due to architectural requirement for single replica operation
```
@eriqua we have a known issue in the application which needs the min replica & max replica setting to 1. Please let us know how we can suppress this issue for this specific module.

eriqua replied:
Hey @AjitPadhi-Microsoft if required for these 2 modules only, please remove this change and update the file

`utilities\pipelines\staticValidation\psrule\.ps-rule\storage-firewall-suppress.Rule.yaml`

adding the following lines

```
---
# Synopsis: Suppress Rules for Multi-Agent Custom Automation Engine module due to architectural requirement for single replica operation
apiVersion: github.com/microsoft/PSRule/v1
kind: SuppressionGroup
metadata:
  name: "SuppressMinReplicas"
spec:
  rule:
    - Azure.ContainerApp.MinReplicas
  if:
    source: Template
    withinPath:
      - avm/ptn/sa/multi-agent-custom-automation-engine/
      - avm/ptn/sa/conversation-knowledge-mining/
```

Not a miss on your end, this is not yet documented. Thank you

eriqua replied:
@AjitPadhi-Microsoft I’ve approved the other PR.
Before approving this one, we first need to merge the other PR, and then you’ll need to pull the latest changes into this branch. Both PRs modify the new PSRule suppress‑rule file.

AjitPadhi-Microsoft committed:
Merge branch 'main' into macae-avm-update

AjitPadhi-Microsoft committed:
psrule issue fix

Commit:
fix: Add suppression rule for Container App min replicas for MACAE & CKM

Commit:
fix: Added suppression related to MACAE alone

eriqua committed:
Merge branch 'main' into macae-avm-update

eriqua commented on utilities/pipelines/staticValidation/psrule/.ps-rule/container-app-minreplica-suppress.Rule.yaml:
```diff
@@ -0,0 +1,13 @@
+---
+# Synopsis: Suppress Rules for Resources that cannot activate firewalls
```
```suggestion
# Synopsis: Suppress Rules for Multi-Agent Custom Automation Engine module due to architectural requirement for single replica operation
```

Prajwal-Microsoft replied:
@eriqua ,

This is updated

Commit:
Modify synopsis for suppression rule

Updated synopsis to reflect the architectural requirement for single replica operation in the Multi-Agent Custom Automation Engine module.

Commit:
Merge branch 'main' into macae-avm-update

Commit:
Merge branch 'main' into macae-avm-update

Commit:
Merge branch 'main' into macae-avm-update

Commit:
Merge branch 'main' into macae-avm-update

Commit:
Merge branch 'main' into macae-avm-update

eriqua committed:
Merge branch 'main' into macae-avm-update

eriqua reviewed (approved):
🆗
